driver.go

```diff
@@ -449,72 +474,101 @@ func (d *Driver) StartTask(cfg *drivers.TaskConfig) (*drivers.TaskHandle, *drive
 
 		for _, port := range allPorts {
-			hostPort := port.Value
+			hostPort := uint16(port.Value)
 			// By default we will map the allocated port 1:1 to the container
-			containerPort := port.Value
+			containerPort := uint16(port.Value)
 
 			// If the user has mapped a port using port_map we'll change it here
 			if mapped, ok := driverConfig.PortMap[port.Label]; ok {
-				containerPort = mapped
+				containerPort = uint16(mapped)
 			}
 
 			// we map both udp and tcp ports
-			publishedPorts = append(publishedPorts, fmt.Sprintf("%s:%d:%d/tcp", network.IP, hostPort, containerPort))
-			publishedPorts = append(publishedPorts, fmt.Sprintf("%s:%d:%d/udp", network.IP, hostPort, containerPort))
+			publishedPorts = append(publishedPorts, apiclient.PortMapping{
+				HostIP:        network.IP,
+				HostPort:      hostPort,
+				ContainerPort: containerPort,
+				Protocol:      "tcp",
+			})
+			publishedPorts = append(publishedPorts, apiclient.PortMapping{
+				HostIP:        network.IP,
+				HostPort:      hostPort,
+				ContainerPort: containerPort,
+				Protocol:      "udp",
+			})
 		}
-
-		createOpts.Publish = &publishedPorts
+		createOpts.ContainerNetworkConfig.PortMappings = publishedPorts
 	}
+	// -------------------------------------------------------------------------------------------
 
 	containerID := ""
 	recoverRunningContainer := false
 	// check if there is a container with same name
-	otherContainerPs, err := d.podmanClient.PsByName(containerName)
+	otherContainerInspect, err := d.podmanClient2.ContainerInspect(d.ctx, containerName)
 	if err == nil {
 		// ok, seems we found a container with similar name
-		if otherContainerPs.State == "running" {
+		if otherContainerInspect.State.Running {
 			// it's still running. So let's use it instead of creating a new one
-			d.logger.Info("Detect running container with same name, we reuse it", "task", cfg.ID, "container", otherContainerPs.Id)
-			containerID = otherContainerPs.Id
+			d.logger.Info("Detect running container with same name, we reuse it", "task", cfg.ID, "container", otherContainerInspect.ID)
+			containerID = otherContainerInspect.ID
 			recoverRunningContainer = true
 		} else {
 			// let's remove the old, dead container
-			d.logger.Info("Detect stopped container with same name, removing it", "task", cfg.ID, "container", otherContainerPs.Id)
-			if err = d.podmanClient.ForceRemoveContainer(otherContainerPs.Id); err != nil {
+			d.logger.Info("Detect stopped container with same name, removing it", "task", cfg.ID, "container", otherContainerInspect.ID)
+			if err = d.podmanClient2.ContainerDelete(d.ctx, otherContainerInspect.ID, true, true); err != nil {
 				return nil, nil, nstructs.WrapRecoverable(fmt.Sprintf("failed to remove dead container: %v", err), err)
 			}
 		}
 	}
 
 	if !recoverRunningContainer {
-		containerID, err = d.podmanClient.CreateContainer(createOpts)
+		// FIXME: there are more variations of image sources, we should handle it
```

Comment:
can we file an issue to track this

Reply:
See https://github.com/hashicorp/nomad-driver-podman/issues/69
I will reference it also in the comment